### PR TITLE
[cleanup] Remove Lexer's unneeded dependency on module.h

### PR DESCRIPTION
**Pull request by yebblies**
Also remove the `Module` constructor from `Loc`.

@WalterBright

### CODE REVIEW

File: src/lexer.c [hunk 1]
```diff
@@ -2353,10 +2351,10 @@ void Lexer::poundLine()
                 continue;                       // skip white space
 
             case '_':
-                if (mod && memcmp(p, "__FILE__", 8) == 0)
```
Comment (AndrejMitrovic) [inline src/lexer.c]: I just hope it doesn't introduce segfaults when printing `__FILE__` when a module declaration is missing (I remember a similar bug with `__MODULE__`).

**Comment by AndrejMitrovic** [comment]
Auto-merge toggled on

File: src/lexer.c (hunk 1 above)
Comment (yebblies) [inline src/lexer.c]: As far as I could tell, it's not possible.

**Comment by yebblies** [comment]
@RazvanN7

**Pull request merged**